modules/linear_boltzmann_solvers/discrete_ordinates_problem/sweep/fluds/aahd_fluds.cu

```diff
@@ -2,353 +2,372 @@
 // SPDX-License-Identifier: MIT
 
 #include "modules/linear_boltzmann_solvers/discrete_ordinates_problem/sweep/fluds/aahd_fluds.h"
-#include "modules/linear_boltzmann_solvers/discrete_ordinates_problem/sweep/fluds/aah_fluds.h"
+#include <cstring>
+#include <numeric>
+
+#include "framework/runtime.h"
+#include "framework/logging/log.h"
 
 namespace opensn
 {
 
-template <typename T>
-inline static char*
-write_to(char* buffer, const T& src)
+static inline bool
+IsTotalSizeNotEmpty(const std::vector<std::size_t>& sizes)
```

Comment:
1. `IsTotalSizeNonZero` might be a better name. Size does not have an empty property, container does.

2. Also, consider changing this into `TotalSizeIsZero` (i.e. inverting the logic) and then user code doing:
```
if (not TotalSizeIsZero(...))
```
which usually reads easier than conditions in negation...

Either solution is fine, your call...